src/main/java/com/github/maxopoly/MemeMana/MemeManaDAO.java

```diff
@@ -42,10 +49,65 @@ public void addManaUnit(MemeManaUnit unit, MemeManaOwner owner) {
 			addManaUnit.setDouble(2, unit.getOriginalAmount());
 			addManaUnit.setDouble(3, unit.getFillGrade());
 			addManaUnit.setDate(4, new Date(unit.getGainTime()));
-			addManaUnit.setInt(5, owner.getID());
+			addManaUnit.setLong(5, owner.getID());
 			addManaUnit.execute();
 		} catch (SQLException e) {
 			logger.log(Level.WARNING, "Problem adding mana unit", e);
 		}
 	}
+
+	public Map<Long,MemeManaPouch> getManaPouches() {
+		try (Connection connection = getConnection();
+				PreparedStatement getManaPouches = connection
+						.prepareStatement("select * from manaUnits;")) {
+			ResultSet rs = getManaPouches.executeQuery();
+			Map<Long,MemeManaPouch> out = new HashMap<Long,MemeManaPouch>();
+			while(rs.next()) {
+				MemeManaUnit unit = new MemeManaUnit(rs.getInt(1),rs.getDouble(2),rs.getDate(4).getTime(),rs.getDouble(3));
+				out.putIfAbsent(rs.getLong(5),new MemeManaPouch(new ArrayList<MemeManaUnit>()));
+				insertChron(unit,out.get(rs.getLong(5)).getUnits());
+			}
+			return out;
+		} catch (SQLException e) {
+			logger.log(Level.WARNING, "Problem getting mana pouches", e);
+			return null;
+		}
+	}
+
+	private void insertChron(MemeManaUnit unit,List<MemeManaUnit> units) {
```

Comment:
Minor complaint, but if you make methods like this, either give them an easier to understand name like "insertManaUnitChronologically" or write a short java doc. Can't know what this method does without reading it, my first reaction was "what is a chron and why does he insert it here"